### ISSUE TITLE
JAWS reading just first line of Iframe content and then stopping after skip to main content

### ISSUE THREAD

**Issue by cksachdev** [FreedomScientific/VFO-standards-support#362]
### Summary
Scenario:
In my application, content in Iframe is loading from server side and entire markup comes from the backend. There is navigation menu buttons in the main html page and Skip to main content link. Role for the iframe body is given as "application"(also tried with document). When Skip to main content is removed, the JAWS read the iframe content as expected, but as soon as "Skip to main content" section is enabled, JAWS reads just the first line inside iframe. What aria role or property I should add, such that when focus is moved from "skip to main content" to iframe, JAWS should start reading the entire content inside iframe? enter image description here Sample code used to move focus inside iframe:
```
    <div class="skipnav"><a tabindex="0" onclick="window.skipNav(event);" onkeydown="window.skipNav(event);">Skip to main content</a></div>

```
```
window.skipNav = function(event) {
    switch (event.type) {
        case "keydown":
            if (event && event.keyCode == 13) {
                setFocusIframe();
                //setTimeout(setFocusIframe, 100);	
            }
            break;
        case "click":
            setFocusIframe();
            //setTimeout(setFocusIframe, 100);
            break;
        default:
            break;

    }

}

function setFocusIframe() {
    // this.iframe[0].contentWindow.document.body.focus()
    var f = jQuery("iframe:first")


    if (f.length > 0) {
        var frameDoc = f[0].contentWindow.document || f[0].contentDocument;
        //f[0].contentWindow.document.body.focus();
        frameDoc.focus();
        console.log("skipNav-------------------------------------------------------->>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>");
    }
}
```

### Expected result
On moving the focus to iframe, JAWS should read the iframe contents.

### Actual result
JAWS reads just the first line and then stops.

#### JAWS version and build number
2020.1912.11 ILM

 
#### Operating System and version
Windows 10

#### Browser and version:
IE 11.592.18362.0

**Comment by JAWS-test2** [comment]
I don't think this is a mistake, but rather the normal behavior of JAWS. If you want to read the rest of the content line by line, you can use the arrow keys or INSERT+DOWN ARROW to read the entire content.

Note that role=application disables the virtual cursor. This ARIA role should not be used for Web documents but only for interactive components that require forms mode for operation and for which there are no ARIA roles.

**Comment by cksachdev** [comment]
@JAWS-test2 Thanks for your response. After reading the first line, there is no indication for the user, that more content is available. Is there any role which can be applied to announce/read the entire content when the focus is moved to iframe body?

**Comment by JAWS-test2** [comment]
I think screen reader users assume that the page contains more content if they have used a skip link. There is no ARIA role to point it out.

I suspect that the problem can be solved by setting the focus not directly on the iframe, but on an element in front of it or an element within the iFrame.

**Comment by cksachdev** [comment]
@JAWS-test2 Because the content in my iframe is dynamic and not known, I will try setting the focus to the first element inside iframe body. Will update if this resolves the problem.

**Comment by JeremiahRogersVispero** [comment]
We would not expect JAWS to read the entire content of the iframe once an element within that iframe is focused. JAWS should read some text when the new item receives focus, depending on how much text that element contains. We would expect the user to deduce, based on context and content, whether there is more to read after JAWS stops reading.
